src/surface_conditions/surface_conditions.jl

```diff
@@ -5,138 +5,70 @@ Updates the value of `p.sfc_conditions` based on the current state `Y` and time
+    sfc_temp_var =
+        p.atmos.surface_model isa PrognosticSurfaceTemperature ?
+        (; sfc_prognostic_temp = Fields.field_values(Y.sfc.T)) : (;)
+    @. sfc_conditions_values = surface_state_to_conditions(
+        surface_state(
+            wrapped_sfc_setup,
             sfc_local_geometry_values,
-            int_ts_values,
-            projected_vector_data(CT1, int_u_values, int_local_geometry_values),
-            projected_vector_data(CT2, int_u_values, int_local_geometry_values),
             int_z_values,
-            params,
-            atmos,
-        )
-    else # SurfaceSetup is a SurfaceState
-        tup_sfc_setup = tuple(sfc_setup)
-        if (is_prognostic_sfc)
-            @. sfc_conditions_values = surface_state_to_conditions(
-                tup_sfc_setup,
-                sfc_local_geometry_values,
-                int_ts_values,
-                projected_vector_data(
-                    CT1,
-                    int_u_values,
-                    int_local_geometry_values,
-                ),
-                projected_vector_data(
-                    CT2,
-                    int_u_values,
-                    int_local_geometry_values,
-                ),
-                int_z_values,
-                params,
-                atmos,
-                sfc_prognostic_temp,
-            )
-        else
-            @. sfc_conditions_values = surface_state_to_conditions(
-                tup_sfc_setup,
-                sfc_local_geometry_values,
-                int_ts_values,
-                projected_vector_data(
-                    CT1,
-                    int_u_values,
-                    int_local_geometry_values,
-                ),
-                projected_vector_data(
-                    CT2,
-                    int_u_values,
-                    int_local_geometry_values,
-                ),
-                int_z_values,
-                params,
-                atmos,
-            )
-        end
-    end
+            t,
+        ),
+        sfc_local_geometry_values,
+        int_ts_values,
+        projected_vector_data(CT1, int_u_values, int_local_geometry_values),
+        projected_vector_data(CT2, int_u_values, int_local_geometry_values),
+        int_z_values,
+        params,
+        atmos,
+        sfc_temp_var...,
+    )
     return nothing
 end
 
+# default case
+sfc_setup_wrapper(sfc_setup::SurfaceState) = (sfc_setup,)
+
+# case when surface setup is func
+sfc_setup_wrapper(sfc_setup::Function) = (sfc_setup,)
+
+#this is the case for the coupler
+function sfc_setup_wrapper(sfc_setup::Fields.Field)
+    @assert eltype(sfc_setup) <: SurfaceState
+    return Fields.field_values(sfc_setup)
+end
+
+surface_state(sfc_setup_wrapper::SurfaceState, _, _, _) = sfc_setup_wrapper
+
+surface_state(
+    sfc_setup_wrapper::Function,
+    sfc_local_geometry_values,
+    int_z_values,
+    t,
+) = sfc_setup_wrapper(sfc_local_geometry_values.coordinates, int_z_values, t)
+
```

Comment:
I'm a bit confused how the dispatch works here. Could you explain a bit more? (Feel free to merge, I just want to make sure I understand).

Reply:
Of course! Because we cannot store a function in a variable this surface_state() function is used in the case where surface setup is a function (the Rico simulation for example) and it passes a changed surface_setup at the last moment as a parameter to the surface_state_to_conditions() function. 

In the other two cases: the coupler case and the default case we change/ wrap the surface setup using the sfc_setup_wrapper() function and store them using a variable wrapped_sfc_setup.

I'm not sure if that clarifies what it's doing?

Reply:
I guess I'm confused about what the function `sfc_setup_wrapper(sfc_local_geometry_values.coordinates, int_z_values, t)` is?

Reply:
I believe that was in the code before in a slightly different format, I'm not 100% sure how it works. I think it casts the default surface setup type to each coordinate, but I don't really know how. I think Dennis probably knows.

Reply:
Oh also it might be unclear that the variable passed into the surface_setup() function has the same name as a complete different function.

Reply:
Yeah that's probably why I am confused I think. I can ask Dennis tomorrow, thanks!

Reply:
Do you want me to change the variable name in another PR and merge that quickly?

Reply:
That might be good, thanks!